x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/PutEngineAction.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.engine.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.DocWriteResponse;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.entsearch.engine.Engine;
+
+import java.io.IOException;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class PutEngineAction extends ActionType<PutEngineAction.Response> {
+
+    public static final PutEngineAction INSTANCE = new PutEngineAction();
+    public static final String NAME = "cluster:admin/engine/put";
+
+    public PutEngineAction() {
+        super(NAME, PutEngineAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest {
+
+        private final Engine engine;
+        private XContentType xContentType = XContentType.JSON;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
```

Comment:
I'm very interested in knowing whether this way of handling the request and extracting the Engine is correct.

Reply:
Makes sense, changed it!